libraries/botbuilder-dialogs-adaptive/src/conditions/onCondition.ts

```diff
@@ -100,6 +100,25 @@ export class OnCondition {
         return [this.onCreateChangeList(planningContext)];
     }
 
+    /**
+     * Get child dialog dependencies so they can be added to the containers dialogset.
+     */
+    public getDependencies(): Dialog[] {
+        const dependencies: Dialog[] = [];
+
+        for (let i = 0; i < this.actions.length; i++) {
+            const action = this.actions[i];
+            dependencies.push(action);
+
+            // Automatically add any child dependencies the dialog might have
+            if (typeof ((action as any) as DialogDependencies).getDependencies == 'function') {
+                ((action as any) as DialogDependencies).getDependencies().forEach((child) => dependencies.push(child));
+            }
+        }
+
+        return dependencies;
+    }
```

Comment:
All you need to do here is "return this.actions;"  When a Dialog is added to a DialogSet it will call getDependencies() recursively to get any of the actions dependencies and any of the dependencies dependencies.